standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/jdbc/functions/HeartbeatTxnRangeFunction.java

```diff
@@ -82,7 +82,7 @@ public HeartbeatTxnRangeResponse execute(MultiDataSourceJdbcResource jdbcResourc
     }
     if (updateCnt == numTxnsToHeartbeat) {
       //fast pass worked, i.e. all txns we were asked to heartbeat were Open as expected
-      context.rollbackToSavepoint(savePoint);
+      context.createSavepoint();
```

Comment:
Dont you need to roll it back?

Reply:
savepoint is kinda commit, same thing is done after refactor in heartbeatTxn/Locks

Reply:
FYI: I removed using `getEpochFn(jdbcResource.getDatabaseProduct())` from this change and switched back to the original way of calculating the timestamp, for two reasons: 
- jdbc paramters are typed. And the framework knows the type of the columns. And I cannot use `getEpochFn(jdbcResource.getDatabaseProduct())` in a parmeterized query as for that column, it expects a bigint value (long in java side). And it doesn't run at all. The possible way to introduce it would be to use it as string, not a paramterized query. 
- the other reason is that I checked other places when we use that column and I wanted to keep the code consistent: Change it at all places or not change at all. 

I would consider doing it at an other, refactor ticket that I mentioned in my previous comment.